iothub/service/src/ConnectionLeaseTimeoutHandler.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// HttpClient delegating handler for periodically revoking the connection lease after a specified amount of time. This allows cached HttpClients to periodically do DNS lookups
+    /// which keeps the client communicating with the correct service endpoint in failover scenarios.
+    /// </summary>
+    internal class ConnectionLeaseTimeoutHandler : DelegatingHandler
```

Comment:
Track 2 doesn't have a class defined like this. Instead they use the ServicePointManager class to set a connection lease timeout. For the life of me though, I couldn't get it to work. Since this approach only sets the connection lease timeouts for our own clients rather than for all http clients directed at the same host, I actually prefer this approach anyways though

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/2c338d26571f4ee84c3c280984af49bcb771fe2a/sdk/core/Azure.Core/src/Pipeline/ServicePointHelpers.cs#L42

Reply:
Did you try asking Pavel for help with that?